src/actions/guides/database/querying.cr

```diff
@@ -7,294 +7,428 @@ class Guides::Database::Querying < GuideAction
+    UserQuery.new.name.ilike("jim")
+    ```
 
-    ### Int32 and Time
+    ### Custom WHERE
 
-    * `gt`
-    * `gte`
-    * `lt`
-    * `lte`
+    Avram also has a `where` method that gives you the ability to do custom and/or programmatic queries.
 
-    ### Examples
+    `WHERE users.name = 'Dan' AND created_at > '#{3.days.ago}' AND created_at < '#{1.hour.ago}'`
 
     ```crystal
-    UserQuery.new.age.gte(18).name.lower.is("sally")
+    UserQuery.new
+      .where(:name, "Dan")
+      .where("created_at > '?' AND created_at < '?'", 3.days.ago, 1.hour.ago)
     ```
 
-    ## Using scopes
+    ## Ordering
 
-    Generally it's best to name your query methods so they are easy to understand
-    and reuse across your app. In Lucky, this is done by extracting methods on the
-    query object.
+    Return rows ordered by the `age` column in descending (or ascending) order.
 
-    Here's an example:
+    `SELECT COLUMNS FROM users ORDER BY users.age DESC`
 
     ```crystal
-    class UserQuery < User::BaseQuery
-      def adults
-        age.gte(18)
-      end
+    UserQuery.new.order_by(:age, :desc)
+    ```
 
-      def search(name)
-        ilike("\#{name}%")
-      end
-    end
+    > The `order_by` method will accept `:asc`, and `:desc` as the ordering direction.
 
-    UserQuery.new.adults.search("Sal")
+    ## Pagination
+
+    To do paginating, you'll use a combination of limit and offset. You can also use this formula to help.
+
+    ```crystal
+    page = 1
+    per_page = 12
+
+    limit = per_page
+    offset = per_page * (page - 1)
+
+    UserQuery.new.limit(limit).offset(offset)
     ```
 
-    ## Preloading associations
+    ### Limit
+
+    `SELECT COLUMNS FROM users LIMIT 1`
+
+    ```crystal
+    UserQuery.new.limit(1)
+    ```
+
+    ### Offset
+
+    `SELECT COLUMNS FROM users OFFSET 20`
+
+    ```crystal
+    UserQuery.new.offset(20)
+    ```
 
-    In development and test environemnts Lucky requries preloading associations.
-    If you forget to preload an association, a runtime error will be raised when
-    you try to access it. In production, the association will be lazy loaded so
-    that users do not see errors.
+    ## Aggregates
 
-    This solution means you will find N+1 queries in development and test, and
-    users will never see an error in production.
+    ### Count
 
-    To preload, just call `preload_{association name}` on the query:
+    `SELECT COUNT(*) FROM users`
 
     ```crystal
-    PostQuery.new.preload_comments
+    total_count = UserQuery.new.select_count
     ```
 
-    ### Customizing how associations are preloaded
+    ### Avg / Sum
 
-    Sometimes you want to order preloads, or add where clauses. To do this, use
-    the `preload_{{ association_name }}` method on the query and pass a query
-    object for the association.
+    `SELECT AVG(users.age) FROM users`
 
     ```crystal
-    PostQuery.new.preload_comments(CommentQuery.new.published(true))
+    UserQuery.new.select_average(:age)
```

Comment:
I think you can use a block like: select_avg(&.age) for type safety. But if not then we should add that to Avram hah

Reply:
Actually it is `column.select_min/max`. See https://github.com/luckyframework/avram/blob/5d7a5090d37a702ff7b7f5b59fb602e4d29dc6e5/spec/query_spec.cr#L312-L382

Reply:
Ah, is that not these methods https://github.com/luckyframework/avram/blob/master/src/avram/query_builder.cr#L123-L137 then?

Reply:
Oops I wasn't clear. What you documented is correct and will work, but there are type-safe versions that should be preferred. Those are the ones I linked to. The method definitions arehttps://github.com/luckyframework/avram/blob/8805b31be56f39976fa8264f3e99aec9fd51b239/src/avram/criteria.cr#L63-L81

Most of the non-type safe versions are available but should only be used if your query is doing something out of the ordinary. The type-safe ones call the non type-safe ones under the hood, but since they are called on a Column's criteria we know the column exists

Reply:
ah. Ok. In that case, maybe I should remove the docs on the `where` that uses the non-type safe call.

Reply:
Actually I think that one is handy in case you need something Avram doesn't support. Like if you need to do a substring match or one of the JSON methods or something. But maybe it belongs in the `Raw SQL` section and the section here should just link to that `### Raw where` -> `"Sometimes Avram doesn't give you exactly what you need. Take a look at the Raw SQL guide to learn how to write any query you need"`